client_side/.infrastructure/infrastructure.sh

```diff
@@ -0,0 +1,201 @@
+##############################################################################
+# This file contains utility functions for use by the interface:
+# - Creating the mock file system for the user
+# - Timing each task
+# - Determining which task and treatment ordering for the current experiment
+# - Determine the next task to move onto and whether the experiment is over
+# - Verify the output of a task
+##############################################################################
+
+# Converts a numeric ASCII value to a character.
+chr() {
+  [ "$1" -lt 256 ] || return 1
+  printf "\\$(printf '%03o' "$1")"
+}
+
+# Converts a character to its ASCII value.
+ord() {
+  LC_CTYPE=C printf '%d' "'$1"
+}
+
+# Outputs the nth alphabetic character. n is 1-based; that is, char_from(1) is "a".
+# $1: the number n specifying which characters
+char_from() {
+  local num_a=$(ord "a")
+  local num_fr=$((num_a + $1 - 1))
+
+  echo $(chr ${num_fr})
+}
+
+# Gets the true task code from the user task number and the task set.
+# The user task number is always sequential.
+# The task can be in either task set 1 or 2, with the dividing task being
+# TASKS_SIZE / 2.
+# That is, if the current task set is 2 and the user current user task number
```

Comment:
This seems inconsistent with the code.  We should be in the else branch because task_set is 2. In the else branch, no subtraction is done so this cannot return a value less than the current user task number.